src/NHibernate/Criterion/ProjectionList.cs

```diff
@@ -14,17 +14,28 @@ public class ProjectionList : IProjection
 	{
 		private IList<IProjection> elements = new List<IProjection>();
 
-		protected internal ProjectionList()
```

Comment:
I'm not sure about need for this option, just a habit from my experience, to try to make all optimization features optional, in case  there are scenarios when they worsen performance. If such a habit is not used in NHibernate project - i guess i'll just remove it and make caching the only behavior.